src/Tribe/I18n.php

```diff
@@ -0,0 +1,210 @@
+ */
+
+namespace Tribe\Events;
+
+use Tribe__Cache_Listener as Cache_Listener;
+use Tribe__Events__Main as TEC;
+use Tribe__Main as Common;
+
+/**
+ * Class I18n
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events
+ */
+class I18n {
+	/**
+	 * An instance of the The Events Calendar main class.
+	 *
+	 * @since TBD
+	 *
+	 * @var TEC
+	 */
+	protected $tec;
+
+	/**
+	 * I18n constructor.
+	 *
+	 * @param TEC $tec An instance of The Events Calendar main class.
+	 */
+	public function __construct( TEC $tec ) {
+		$this->tec = $tec;
+	}
+
+	/**
+	 * Get all possible translations for a String based on the given Languages and Domains
+	 *
+	 * WARNING: This function is slow because it deals with files, so don't overuse it!
+	 *
+	 * @since TBD Moved here from Tribe__Events__Main.
+	 *
+	 * @param array  $strings          An array of strings (required).
+	 * @param array  $languages        Which l10n to fetch the string (required).
+	 * @param array  $domains          Possible Domains to re-load.
+	 * @param string $default_language The default language to avoid re-doing that.
+	 *
+	 * @return array                    A multi level array with the possible translations for the given strings
+	 */
+	public function get_i18n_strings( $strings, $languages, $domains = array(), $default_language = 'en_US' ) {
+		$domains = wp_parse_args( $domains, array(
+			'default'             => true, // Default doesn't need file path
+			'the-events-calendar' => $this->tec->plugin_dir . 'lang/',
+		) );
+
+		return $this->get_i18n_strings_for_domains( $strings, $languages, $domains );
+	}
+
+	/**
+	 * Get all possible translations for a String based on the given Languages and Domains
+	 *
+	 * WARNING: This function is slow because it deals with files, so don't overuse it!
+	 * Differently from the `get_i18n_strings` method this will not use any domain that's not specified.
+	 *
+	 * @since TBD
+	 *
+	 * @todo Include support for the `load_theme_textdomain` + `load_muplugin_textdomain`
+	 *
+	 *
+	 * @param array $strings   An array of strings (required)
+	 * @param array $languages Which l10n to fetch the string (required)
+	 * @param array $domains   Possible Domains to re-load
+	 *
+	 * @return array                    A multi level array with the possible translations for the given strings
+	 */
+	public function get_i18n_strings_for_domains( $strings, $languages, $domains = array( 'default' ) ) {
+		sort( $languages );
+		$strings_buffer = [ $strings ];
+
+		foreach ( $languages as $language ) {
+			// Override the current locale w/ the one we need to compile the translations.
+			$language_strings = $this->with_locale(
+				$language,
+				[ $this, 'compile_translations' ],
+				[ $strings, $domains ]
+			);
+			$strings_buffer[] = $language_strings;
+		}
+
+		$strings = count( $strings_buffer ) > 1 ? array_merge_recursive( ...
+			$strings_buffer ) : reset( $strings_buffer );
+
+		// Prevent Empty Strings and Duplicates
+		foreach ( $strings as $key => $value ) {
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 6210976e37401e713dec7aefb1729d9d</sub>